rules/reports/reports.rules

```diff
@@ -0,0 +1,72 @@
+# -*- mode: Snakemake -*-
+#
+# ReportGeneration rules
+
+import pandas
+from io import StringIO
+
+rule all_reports:
+    input:
+        TARGET_REPORT
+
+def parse_trim_summary(filename):
+    with open(filename) as f:
```

Comment:
If you're opening a file as your first statement, I prefer writing a function that takes a file object.  You can open the file just before calling the function.  This way, the function is easier to test (you don't need to make a temp file for testing).